bpf/lib/process.h

```diff
@@ -311,10 +312,16 @@ typedef __u64 mbset_t;
 struct binary {
 	// length of the path stored in path, this should be < BINARY_PATH_MAX_LEN
 	// but can contain negative value in case of copy error.
-	// While s16 would be sufficient, 64 bits are handy for alignment.
-	__s64 path_length;
+	// While s16 would be sufficient, 32 bits are handy for alignment.
+	__s32 path_length;
+	// if end_r contains reversed path postfix
+	__u32 reversed;
 	// BINARY_PATH_MAX_LEN first bytes of the path
 	char path[BINARY_PATH_MAX_LEN];
+	// BINARY_PATH_MAX_LEN last bytes of the path
+	char end[BINARY_PATH_MAX_LEN];
+	// BINARY_PATH_MAX_LEN reversed last bytes of the path
+	char end_r[BINARY_PATH_MAX_LEN];
 	// matchBinary bitset for binary
```

Comment:
Sorry just noticed that the postfix LPM TRIE maps only go up to STRING_POSTFIX_MAX_LENGTH in size (= 128).
So here should probably be:
```
char end[STRING_POSTFIX_MAX_LENGTH];
char end_r[STRING_POSTFIX_MAX_LENGTH];
```